pkg/multisets/string.go

```diff
@@ -0,0 +1,166 @@
+/*
+Copyright 2020 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+// Package ghclient provides a github client that wraps go-github with retry logic, rate limiting,
+// and depagination where necessary.
+package multisets
+
+import (
+	"math/rand"
+	"sort"
+
+	"k8s.io/apimachinery/pkg/util/sets"
+)
+
+// String is a layered set implemented as a map of sets.String.
+type String map[int]sets.String
```

Comment:
Maybe define a `Set` interface so this doesn't need a concrete implementation per type and can be nested arbitrarily deep?

Reply:
I need to check because I need a Union with a Set... Probably it'll work if I name it differently.
I'll check tomorrow.